staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response_test.go

```diff
@@ -222,3 +227,142 @@ func TestWatchEncoderIdentifier(t *testing.T) {
 		t.Error("  - watchEncoder::typeIdentifier to capture all relevant fields in identifier")
 	}
 }
+
+func TestWatchListEncoder(t *testing.T) {
+	scenarios := []struct {
+		name              string
+		negotiatedEncoder *fakeEncoder
+
+		actualEvent         watch.Event
+		actualListBlueprint runtime.Object
+
+		targetGVK *schema.GroupVersionKind
+
+		validate func(t *testing.T, rawEncoder *fakeEncoder, transformedEvent watch.Event)
+	}{
+		{
+			name:        "pass through, an obj without the annotation received",
+			actualEvent: makeBookmarkEventFor(makePod("1")),
+			validate: func(t *testing.T, negotiatedEncoder *fakeEncoder, transformedEvent watch.Event) {
+				if negotiatedEncoder != nil {
+					t.Fatal("negotiatedEncoder should be nil")
+				}
+				expectedEvent := makeBookmarkEventFor(makePod("1"))
+				cmp.Equal(expectedEvent, transformedEvent)
+			},
+		},
+		{
+			name:                "encodes the initialEventsListBlueprint if an obj with the annotation is passed",
+			actualEvent:         makeBookmarkEventFor(makePodWithInitialEventsAnnotation("1")),
+			actualListBlueprint: makePodList("100"),
+			negotiatedEncoder:   &fakeEncoder{},
+			validate: func(t *testing.T, negotiatedEncoder *fakeEncoder, transformedEvent watch.Event) {
```

Comment:
I don't like it. We don't want to duplicate the actual code in tests.
The tests should have value of the annotation explicitly defined, instead of trying it compute it.

So we should change the test by removing the `validate` function and instead adding a field:
`expectedAnnotation string`

And in this case simply doing sth like
```
expectedAnnotation: base64.Encode(negotiatedEncoder(&v1.PodList{}, corev1))
```

Reply:
done.